src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -23,10 +29,29 @@ internal JsonElement(JsonDocument parent, int idx)
             _idx = idx;
         }
 
-        internal JsonTokenType TokenType => _parent?.GetJsonTokenType(_idx) ?? JsonTokenType.None;
+        private JsonTokenType TokenType => _parent?.GetJsonTokenType(_idx) ?? JsonTokenType.None;
 
+        /// <summary>
+        ///   The <see cref="JsonValueType"/> that the value is.
+        /// </summary>
+        /// <exception cref="ObjectDisposedException">
+        ///   The parent <see cref="JsonDocument"/> has been disposed.
+        /// </exception>
         public JsonValueType Type => TokenType.ToValueType();
 
+        /// <summary>
+        ///   Get the value at a specified index when the current value is a
+        ///   <see cref="JsonValueType.Array"/>.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        ///   This value's <see cref="Type"/> is not <see cref="JsonValueType.Array"/>.
+        /// </exception>
+        /// <exception cref="IndexOutOfRangeException">
```

Comment:
❔  Is this correct? If the arguments are validated, we would typically expect to see `ArgumentOutOfRangeException` for cases less than 0, and `ArgumentException` for cases where it is larger than the length. `IndexOutOfRangeException` is not an `ArgumentException` and is more commonly associated with the `ldelem` and `stelem` IL instructions.